### PR TITLE
Keep pixel index value as instance variable for CMYK image

### LINKED ISSUE #747: Can't get black value of a pixel in a CMYK Image
ImageMagick 6.9.10-49
I tried ruby 2.5.1 and ruby 2.6.1
I tried RMagick 3.2.0, 3.1.0 and 2.16.0

I try to read the black value of some pixels in a CMYK image without luck.

I always get back `0`

I checked ImageMagick with this command:

```bash
$ convert black.jpg -format '%[pixel:p{1,1}]' info:-
cmyk(0,0,0,255)%
```

When I try to get the same value with RMagick with this script

```ruby
require 'rmagick'

black = Magick::Image.read('black.jpg')[0]
black_pixels = []
black.each_pixel do |pixel|
    black_pixels << pixel
end
puts "black.jpg Pixel 1/1: cmyk(#{black_pixels.first.cyan},#{black_pixels.first.magenta},#{black_pixels.first.yellow},#{black_pixels.first.black}),"

cyan = Magick::Image.read('cyan.jpg')[0]
cyan_pixels = []
cyan.each_pixel do |pixel|
    cyan_pixels << pixel
end
puts "cyan.jpg Pixel 1/1: cmyk(#{cyan_pixels.first.cyan},#{cyan_pixels.first.magenta},#{cyan_pixels.first.yellow},#{cyan_pixels.first.black})"
```

I get these results:

```bash
ruby read_color.rb
black.jpg Pixel 1/1: cmyk(0,0,0,0),
cyan.jpg Pixel 1/1: cmyk(65278,0,0,0)
```
Other color values are working, but not the black value.

I suspect that the error has something to do with the fact that black is associated with 'opacity' in ext/RMagick/rmpixel.c 

```c
DEF_PIXEL_CMYK_CHANNEL_ACCESSOR(cyan, red)
DEF_PIXEL_CMYK_CHANNEL_ACCESSOR(magenta, green)
DEF_PIXEL_CMYK_CHANNEL_ACCESSOR(yellow, blue)
DEF_PIXEL_CMYK_CHANNEL_ACCESSOR(black, opacity)
```
Here is my demo code: [https://github.com/onlinetocode/rmagick_black_test](https://github.com/onlinetocode/rmagick_black_test)

### PR DESCRIPTION
**Pull request by Watson1978**
Fix #747

### CODE REVIEW

File: ext/RMagick/rmpixel.c [hunk 1]
```diff
@@ -633,7 +639,10 @@ Pixel_from_MagickPixel(const MagickPixel *pp)
     pixel->blue    = ROUND_TO_QUANTUM(pp->blue);
     pixel->opacity = ROUND_TO_QUANTUM(pp->opacity);
 
-    return Data_Wrap_Struct(Class_Pixel, NULL, destroy_Pixel, pixel);
+    obj = Data_Wrap_Struct(Class_Pixel, NULL, destroy_Pixel, pixel);
```
Comment (dlemstra) [inline ext/RMagick/rmpixel.c]: I am wondering if Pixel should be changed to use a `QuantumPixelPacket` instead?

Comment (Watson1978) [reply ext/RMagick/rmpixel.c]: Yeah, I’m think so

Comment (onlinetocode) [reply ext/RMagick/rmpixel.c]: I tested this fix. Getting the black value is possible but setting the black value isn't. the `Magick::Pixel#black=` method is gone.

Comment (Watson1978) [reply ext/RMagick/rmpixel.c]: With this way, I think it is difficult to enable `Magick::Pixel#black=` ...

**Comment by Watson1978** [comment]
Created another implementation in #759

**Pull request closed without merging**